lib/ruby_indexer/lib/ruby_indexer/entry.rb

```diff
@@ -136,6 +136,9 @@ class Module < Namespace
     class Class < Namespace
       extend T::Sig
 
+      OBJECT_NESTING = T.let(["Object"].freeze, T::Array[String])
```

Comment:
TIL: using arrays in case statement will allocate a new array each time the case statement is hit.

Script:

```rb
a = ["Object"]
b = ["Object"]
c = []

CONST = ["Object"]

def match_with_const(x)
  case x
  when CONST
    puts "Matched"
  end
end

def match_with_array(x)
  case x
  when ["Object"]
    puts "Matched"
  end
end

puts "Arrays count before match_with_const: #{ObjectSpace.count_objects[:T_ARRAY]}"
match_with_const a
match_with_const b
match_with_const c
puts "Arrays count after match_with_const: #{ObjectSpace.count_objects[:T_ARRAY]}"

puts "Arrays count before match_with_const: #{ObjectSpace.count_objects[:T_ARRAY]}"
match_with_array a
match_with_array b
match_with_array c
puts "Arrays count after match_with_const: #{ObjectSpace.count_objects[:T_ARRAY]}"
```

Output:

```
Arrays count before match_with_const: 1804
Matched
Matched
Arrays count after match_with_const: 1804
Arrays count before match_with_const: 1804
Matched
Matched
Arrays count after match_with_const: 1807
```